code/lib/core-server/src/presets/common-preset.ts

```diff
@@ -194,20 +194,31 @@ export const features = async (
   legacyDecoratorFileOrder: false,
 });
 
-export const storyIndexers: StorybookConfig['storyIndexers'] = async (existingIndexers) => {
-  const csfIndexer: StoryIndexer['indexer'] = async (fileName, opts) => {
-    const code = (await readFile(fileName, 'utf-8')).toString();
-    return loadCsf(code, { ...opts, fileName }).parse();
-  };
-  return [
-    {
-      test: /(stories|story)\.(m?js|ts)x?$/,
-      indexer: csfIndexer,
-    },
-    ...(existingIndexers || []),
-  ];
+export const csfIndexer: Indexer = {
+  test: /\.stories\.(m?js|ts)x?$/,
+  index: async (fileName, options) => {
+    const csf = (await readCsf(fileName, options)).parse();
+
+    // eslint-disable-next-line no-underscore-dangle
+    return Object.entries(csf._stories).map(([exportName, story]) => {
```

Comment:
Can we add this to a getter on `CsfFile`? The code seems to be duplicated all over the place and that would make this PR a lot more DRY.

Reply:
`CsfFile.indexEntries`?

Reply:
Yeah I've been thinking something similar, I'll experiment with a getter for  `indexInputs` and stories with `exportName` so you don't need the funky use of `_stories`.